### PR TITLE
feat: [SSCA-811]: Added Project Params and PipelineExecutionId in Enforcement Summary

### PR DESCRIPTION
**Pull request by arpit-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54560)

### CODE REVIEW

**Review by deepharness — CHANGES_REQUESTED** [review]
> Reviewed 4 of 13 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54560)* status: 9 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`ssca-manager/service/src/main/java/io/harness/repositories/EnforcementSummaryRepoCustomImpl.java` line 47 at r1](https://reviewable.io/reviews/harness/harness-core/54560#-NhZszqzDtq1_fDs4wpk:-NhZszqzDtq1_fDs4wpl:bfwoyly) ([raw file](https://github.com/harness/harness-core/blob/f2b248225eff92ece52826117d0ad49abf6ee351/ssca-manager/service/src/main/java/io/harness/repositories/EnforcementSummaryRepoCustomImpl.java#L47)):*
> ```Java
> 
>   @Override
>   public EnforcementSummaryEntity findOne(Criteria criteria) {
> ```

We should add accountId as you had already done the migration
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/migration/EnforcementSummaryProjectParamMigration.java` line 43 at r1](https://reviewable.io/reviews/harness/harness-core/54560#-NhZsSxUEtek_xuZlImi:-NhZsSxUEtek_xuZlImj:b-ga80qp) ([raw file](https://github.com/harness/harness-core/blob/f2b248225eff92ece52826117d0ad49abf6ee351/ssca-manager/service/src/main/java/io/harness/ssca/migration/EnforcementSummaryProjectParamMigration.java#L43)):*
> ```Java
>             .find(query.getQueryObject());
> 
>     for (Document document : iterable) {
> ```

Also we should do batch update for collection , saving one at  a time will consume more resources on mongo.
https://stackoverflow.com/questions/31826760/how-to-get-data-in-batches-in-mongodb
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/migration/EnforcementSummaryProjectParamMigration.java` line 50 at r1](https://reviewable.io/reviews/harness/harness-core/54560#-NhZpo0u3NqMHUS_W7VG:-NhZpo0v6YyCBx41kf1s:b-4kaams) ([raw file](https://github.com/harness/harness-core/blob/f2b248225eff92ece52826117d0ad49abf6ee351/ssca-manager/service/src/main/java/io/harness/ssca/migration/EnforcementSummaryProjectParamMigration.java#L50)):*
> ```Java
>             Criteria.where(ArtifactEntity.ArtifactEntityKeys.orchestrationId).is(summaryEntity.getOrchestrationId());
> 
>         ArtifactEntity artifact = artifactRepository.findOne(criteria);
> ```

You can keep a fixed size map of orchestrationId -> artifactId , this will reduce the hit on artifact repo. 
https://stackoverflow.com/questions/5601333/limiting-the-max-size-of-a-hashmap-in-java

File: ssca-manager/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -917,13 +917,28 @@ components:
...
+        created:
+          type: number
       required:
         - artifact
         - enforcement_id
         - orchestration_id
         - deny_list_violation_count
         - allow_list_violation_count
         - status
+        - account_id
```
Comment (reetika04) [inline ssca-manager/contracts/openapi/v1/openapi.yaml:937]: We can get account id from the header, we can remove this from the body.

Comment (arpit-harness) [reply ssca-manager/contracts/openapi/v1/openapi.yaml:937]: Body is the whole DTO which we might send back later as responses.

**Review by arpit-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54560)* status: 10 file(s) to review. 1 change requests. 4. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`ssca-manager/service/src/main/java/io/harness/repositories/EnforcementSummaryRepoCustomImpl.java` line 47 at r1](https://reviewable.io/reviews/harness/harness-core/54560#-NhZszqzDtq1_fDs4wpk:-NhlDmboAA6P7cR9x9ZS:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/f2b248225eff92ece52826117d0ad49abf6ee351/ssca-manager/service/src/main/java/io/harness/repositories/EnforcementSummaryRepoCustomImpl.java#L47)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

We should add accountId as you had already done the migration
</blockquote></details>

Done
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/migration/EnforcementSummaryProjectParamMigration.java` line 43 at r1](https://reviewable.io/reviews/harness/harness-core/54560#-NhZsSxUEtek_xuZlImi:-NhlDo1-7GP806CzgqLA:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/f2b248225eff92ece52826117d0ad49abf6ee351/ssca-manager/service/src/main/java/io/harness/ssca/migration/EnforcementSummaryProjectParamMigration.java#L43)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

Also we should do batch update for collection , saving one at  a time will consume more resources on mongo.
https://stackoverflow.com/questions/31826760/how-to-get-data-in-batches-in-mongodb
</blockquote></details>

Done
___
*[`ssca-manager/service/src/main/java/io/harness/ssca/migration/EnforcementSummaryProjectParamMigration.java` line 50 at r1](https://reviewable.io/reviews/harness/harness-core/54560#-NhZpo0u3NqMHUS_W7VG:-NhlDpnmAiQWtn0Zvc1v:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/f2b248225eff92ece52826117d0ad49abf6ee351/ssca-manager/service/src/main/java/io/harness/ssca/migration/EnforcementSummaryProjectParamMigration.java#L50)):*
<details><summary><i>Previously, deepharness wrote…</i></summary><blockquote>

You can keep a fixed size map of orchestrationId -> artifactId , this will reduce the hit on artifact repo. 
https://stackoverflow.com/questions/5601333/limiting-the-max-size-of-a-hashmap-in-java
</blockquote></details>

Done

**Pull request closed without merging**